xonsh/tools.py

```diff
@@ -909,3 +909,50 @@ def check_for_partial_string(x):
         return (string_indices[-1], None, starting_quote[-1])
     else:
         return (string_indices[-2], string_indices[-1], starting_quote[-1])
+
+
+# expandvars is a modified version of os.path.expandvars from the Python 3.5.1
+# source code (root/Lib/posixpath.py, line 266)
+
+_varprog = None
+
+
+def expandvars(path):
```

Comment:
This seems way more complex than using [`Env.replacve_env()`](http://xonsh.org/api/environ.html#xonsh.environ.Env.replace_env) to replaces the contents of os.environ with detyped env, and then call os.expandvars().  Is this doing something more xonsh-specific that I am not seeing? Or is it more performant in some way?